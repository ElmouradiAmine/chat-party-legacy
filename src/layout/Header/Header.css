.header {
  display: grid;
  grid-template-columns:
    [full-start] minmax(0, 1fr) [content-start]minmax(300px, 1024px)
    [content-end]minmax(0, 1fr) [full-end];
  column-gap: var(--grid-gutter-column);

  background-color: var(--color-primary);
  border-bottom: var(--border-primary);
  height: 6.4rem;
}

.header__content {
  position: relative;
  grid-column: content-start / content-end;
  display: flex;
  align-items: center;
  justify-content: space-between;
}
