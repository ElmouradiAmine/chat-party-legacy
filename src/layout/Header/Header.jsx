import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../../components/Logo/Logo';

import './Header.css';
import UsersCount from '../../features/usersCount/UsersCount';

const Header = ({ className }) => (
  <header className={`header ${className}`}>
    <div className="header__content">
      <Logo className="header__logo" />
      <UsersCount />
    </div>
  </header>
);

Header.propTypes = {
  className: PropTypes.string,
};

Header.defaultProps = {
  className: '',
};

export default Header;
