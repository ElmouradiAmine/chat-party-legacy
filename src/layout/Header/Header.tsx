import React from 'react';

import './Header.css';

export interface Props {
  className?: string;
}

const Header: React.FC<Props> = ({ className = '' }) => (
  <header className={`header ${className}`}>
    Header
  </header>
);

export default Header;
