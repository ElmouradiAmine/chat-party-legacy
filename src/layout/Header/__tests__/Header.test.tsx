import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Header, { Props } from '../Header';

afterEach(cleanup);
describe('Header component', () => {
  const { className }: Props = {
    className: 'className',
  };
  test('renders correctly', () => {
    render(<Header />);
  });

  test('renders correctly with props', () => {
    const { container } = render(<Header className={className} />);
    expect(container.firstElementChild?.classList).toContain(className);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Header className={className} />);
    expect(tree).toMatchSnapshot();
  });
});
