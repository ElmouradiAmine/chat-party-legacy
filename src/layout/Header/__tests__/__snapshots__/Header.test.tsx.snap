// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  className="header className"
>
  Header
</header>
`;
